### ISSUE TITLE
Mega Torch does not block Cataclysm's Lionfish spawns

### ISSUE THREAD

**Issue by Kadah** [Xalcon/TorchMaster#223]
Found this behavior in the Direwolf20 1.20 pack and initially reported it there as it initially seemed like it could be resolved with a config change but now appears it cannot. https://github.com/FTBTeam/FTB-Modpack-Issues/issues/4019

The Deepling Angler hostile mob spawns with a pet Lionfish on a lead. The Mega Torch blocks the Deepling Angler's spawn, but not the Lionfish. The result is the ocean ends up full of randomly-hostile Lionfish (some behavior with the Cataclysm mod makes them not always agro, I'm not actually familar with the mod).

![image](https://github.com/Xalcon/TorchMaster/assets/751209/818a6c8d-d2ae-4044-b70c-0116be8d74fb)

**Comment by Xalcon** [comment]
Can you crosscheck if the dreadlamp works?

**Comment by Kadah** [comment]
No change in spawns. 

Noticed the same behavior in the Rats mod's Ratlantis, Mega Torch blocks the Pirate Rats from spawning, but their (not empty) boats continue to appear. Dreadlamp had no affect here as well.

![image](https://github.com/Xalcon/TorchMaster/assets/751209/c6031827-7ef1-4f5e-a032-3f99746ad92f)

**Comment by Xalcon** [comment]
This looks like a bug on their end then. It looks like they try to spawn the enemy, which gets blocked but they don't stop spawning auxiliary entities. This can easily happen when they spawn the boat first.

For the other case, I assume they either spawn the mobs differently or there is an edge case we're my hostile mob detection code doesn't work. I'll look into it

**Comment by Xalcon** [comment]
I've looked into the issue and its pretty much the same as the boat issue. The lionfish is correctly identified as a hostile mob and added to the mega torch. The angler entity attempts its spawning - during the setup period, it spawns the lionfish, which is then attached to the angler entity - but the main entity gets blocked. Unfortunately, the lionfish is spawned without utilizing the correct events.

In other words, the lionfish is spawning because the spawning events are missing, so Torchmaster doesnt see it spawning. It's impossible for me to detect if the lionfish was spawned naturally or via the angler or any other mean. 

I can block the spawning of the lionfish outright, but thats rather invasive, since it would prevent the spawning via spawners as well (for farming purposes).

**Comment by Xalcon** [comment]
I did find an issue in my code! I'll provide a fix in a few minutes, hopefully this fixes both issues.

**Comment by Xalcon** [comment]
This should be fixed in 20.1.5
Available on [Curseforge](https://legacy.curseforge.com/minecraft/mc-mods/torchmaster/files/4817795), [Modrinth ](https://modrinth.com/mod/torchmaster)and the [Github Release Page](https://github.com/Xalcon/TorchMaster/releases/tag/mc1.20.1-20.1.5)

**Comment by Kadah** [comment]
Can confirm its fixed for the two cases I've found so far. Nice to not have dozens of orphaned buddy spawns.
